lib/impl/refactorRules.js

```diff
@@ -0,0 +1,384 @@
+var async = require('async');
+var getSectionMap = require('../utils/get-section-map');
+var _ = require('underscore');
+var models = require('../common/models.js')();
+
+/*
+  Checking rule targets are still valid -- if not, rule is deleted
+*/
+module.exports = function refactorRules(connections, form, ruleDeletionFlags, cb) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Function has a complexity of 105 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This function has 376 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)